calico_node/startup.go

```diff
@@ -51,22 +53,71 @@ func main() {
 	// Make sure we have a global cluster ID set.
 	log.Info("Ensuring a cluster guid is set")
 	ensureClusterGuid(c)
+
+	// Update the Node resource if necessary.
 }
 
 // ensureClusterGuid assigns a cluster GUID if one doesn't exist.
-func ensureClusterGuid(c api.Client) {
+func ensureClusterGuid(c bapi.Client) {
 	guid := hex.EncodeToString(uuid.NewV4().Bytes())
 	_, err := c.Create(&model.KVPair{
 		Key:   model.GlobalConfigKey{Name: "ClusterGUID"},
 		Value: guid,
 	})
 	if err != nil {
 		if _, ok := err.(errors.ErrorResourceAlreadyExists); !ok {
-			log.Warnf("Failed to set ClusterGUID: %s", err)
+			log.WithError(err).Warn("Failed to set ClusterGUID")
 			panic(err)
 		}
 		log.Infof("Using previously configured ClusterGUID")
 		return
 	}
 	log.Infof("Assigned ClusterGUID %s", guid)
 }
+
+// Update the current Node resource configuration (if it exists).
+func updateNodeResource() {
+	var c client.Client
+	var err error
+	var node *api.Node
+	var modified bool
+
+	if cfg, err := client.LoadClientConfig(""); err != nil {
+		panic(fmt.Sprintf("Error loading config: %s", err))
+	} else if c, err = client.Client(cfg); err != nil {
+		panic(fmt.Sprintf("Error creating client: %s", err))
+	}
+
+	// Get the current Node resource configuration.  If it doesn't exist,
+	// initialise it now.
+	nm := api.NodeMetadata{Name: os.Getenv("NODENAME")}
+	if node, err = c.Nodes().Get(nm); err != nil {
+		if ok := err.(errors.ErrorResourceDoesNotExist); !ok {
+			panic(fmt.Sprintf("Error contacting datastore: %s", err))
+		} else {
+			log.WithField("Name", nm.Name).Info("Node has not yet been configured")
+			node = &api.Node{Metadata: nm}
+			modified = true
+		}
+	}
+	log.Infof("Current node config: %v", node)
+
+	// Check to see if we are on AWS, and if so get the IPv4 and IPv6
+	// subnets (if they have not yet been configured).
```

Comment:
@caseydavenport  I have not done interface based querying yet simply because it overlaps too much with the PR that @2blesd2bstresd is working on.   I don't think it makes sense to do that until that PR is in - the general flow will then be sorted out.

That said - the purpose of this current PR is to check our assumptions about what can be used for the IPIP decision.  See my above note about the metadata subnet vs vpc CIDRs